tests/test_other.py

```diff
@@ -1912,20 +1912,30 @@ def test_prepost(self):
     # noInitialRun prevents run
     for no_initial_run, run_dep in [(0, 0), (1, 0), (0, 1)]:
       print(no_initial_run, run_dep)
-      Popen([PYTHON, EMCC, os.path.join(self.get_dir(), 'main.cpp'), '-s', 'BINARYEN_ASYNC_COMPILATION=0']).communicate()
-      src = 'var Module = { noInitialRun: %d };\n' % no_initial_run + open(os.path.join(self.get_dir(), 'a.out.js')).read()
+      args = ['-s', 'BINARYEN_ASYNC_COMPILATION=0']
+      if no_initial_run:
+        args += ['-s', 'INVOKE_RUN=0']
       if run_dep:
-        src = src.replace('// {{PRE_RUN_ADDITIONS}}', '// {{PRE_RUN_ADDITIONS}}\naddRunDependency("test");') \
-                 .replace('// {{POST_RUN_ADDITIONS}}', '// {{POST_RUN_ADDITIONS}}\nremoveRunDependency("test");')
-      open(os.path.join(self.get_dir(), 'a.out.js'), 'w').write(src)
-      assert ('hello from main' in run_js(os.path.join(self.get_dir(), 'a.out.js'))) != no_initial_run, 'only run if no noInitialRun'
+        with open('pre.js', 'w') as f:
+          f.write('Module.preInit = function() { addRunDependency("test"); }')
```

Comment:
Using `preInit` works, however, it would be good to eventually remove it (as we have too many overlapping related options, I think). So I'd prefer not to use it in more places. But I suppose this test would need some larger refactoring for that anyhow, so no objection here.

Reply:
I should just as well use preRun I think.   Would that be preferable?

Reply:
If preRun works, yeah, that sounds great. (It is slightly different though, but I think here it should be good enough.)

lgtm with that.